### PR TITLE
Fix query database drop-down rendering location after resize

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR fixes issue #15499

This PR resolves issue #15499 by correcting the rendering behavior for the select database drop down list when the list is expanded and the ADS shell window is resized. 

Steps to verify fix:
Run a query
Expand the database drop-down
Resize the ADS shell window
Correct & expected behavior: Drop-down doesn't appear at window origin position (0px, 0px) after resize.
Incorrect & previous behavior: Drop down appears at the window origin position (0px, 0px) after resize.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
What was the issue here? Is this actually something that could affect VS Code as well? If so we should look into getting the fix in for them as well.

(in the future it's generally helpful for reviewers to include in the description what the problem was and how you arrived at your fix since people often won't have context. And then you should also include in the comments why you're changing behavior and what it's fixing. This is especially important for VS Code edits since we'll need that context later on if we're dealing with merge conflicts in the code)

**Comment by abist** [comment]
@Charles-Gagnon Just relayed the same thing to him. I'm checking whether this actually needs a change under `vs/` or not

**Comment by lewis-sanchez** [comment]
@Charles-Gagnon, the issue I'm trying to solve is described here: https://github.com/microsoft/azuredatastudio/issues/15499

What I found while looking into the issue was that DOM.getDomNodePagePosition(this.view).top in old line 322:

this.view.style.top = `${top - (this.useFixedPosition ? DOM.getDomNodePagePosition(this.view).top : containerPosition.top)}px`;

was returning the same value as top  and would result in 0px being assigned to this.view.style.top, the same thing was happening on old line 323 where left was being assigned. There's an onboarding video I was linked that mentioned that we like to avoid changing files in the "vs/" because it can result in merge conflicts when we bring in VS code into ADS.

I'm not entirely sure, if we can make the change in "/sql", but that would definitely be preferred over the current approach, but when I was debugging through this, I was only able to find one spot where we were setting left and top values for UI elements. I'm looking into alternatives with Aditya.

**Comment by abist** [comment]
@lewis-sanchez I think the correct fix here would be to fix the underlying issue in `dropdown.ts` which is under `sql` and uses the `List` component from `vscode`. The usual `vscode`dropdown behavior is to hide context views when resizing, but our editable dropdown based off `vscode`'s dropdown doesn't follow this behavior. The correct fix in my opinion would be to add something like this in the `dropdown.ts` file:
```
		this._register(DOM.addDisposableListener(window, DOM.EventType.RESIZE, () => {
			this._hideList();
		}));
```

File: src/sql/base/parts/editableDropdown/browser/dropdown.ts [hunk 1]
```diff
@@ -136,6 +136,10 @@ export class Dropdown extends Disposable implements IListVirtualDelegate<string>
 			}
 		});
 
+		this._register(DOM.addDisposableListener(window, DOM.EventType.RESIZE, () => {
```
Comment (Charles-Gagnon) [inline src/sql/base/parts/editableDropdown/browser/dropdown.ts:139]: It'd be great if you would add a comment explaining why this is necessary

**Review by Charles-Gagnon — APPROVED** [review]

**Review by abist — APPROVED** [review]

**Review by abist — APPROVED** [review]

**Review by kburtram — APPROVED** [review]

**Comment by lewis-sanchez** [comment]
@Charles-Gagnon, is it okay for me to merge this?

**Pull request merged**